tests/bdd/features/volume/resize/test_resize_offline.py

```diff
@@ -0,0 +1,448 @@
+
+
+POOL1_UUID = "91a60318-bcfe-4e36-92cb-ddc7abf212ea"
+POOL2_UUID = "92a60318-bcfe-4e36-92cb-ddc7abf212ea"
+POOL3_UUID = "93a60318-bcfe-4e36-92cb-ddc7abf212ea"
+POOL4_UUID = "94a60318-bcfe-4e36-92cb-ddc7abf212ea"
+VOLUME_UUID = "5cd5378e-3f05-47f1-a830-a0f5873a1449"
+DEFAULT_REPLICA_CNT = 3
+DEFAULT_POOL_SIZE = 209715200
+NODE1_NAME = "io-engine-1"
+NODE2_NAME = "io-engine-2"
+NODE3_NAME = "io-engine-3"
+NODE4_NAME = "io-engine-4"
+VOLUME_CTX_KEY = "volume"
+VOLUME_SIZE = 52428800  # 50MiB
+VOLUME_NEW_SIZE = 104857610  # 100MiB
+VOLUME_SHRINK_SIZE = 41943040  # 40MiB
+
+
+@scenario("resize_offline.feature", "Expand a published volume")
+def test_expand_a_published_volume():
+    """Expand a published volume."""
+
+
+@scenario(
+    "resize_offline.feature",
+    "Expand a published volume after unpublishing it while having an offline replica",
+)
+def test_expand_a_published_volume_after_unpublishing_it_while_having_an_offline_replica():
+    """Expand a published volume after unpublishing it while having an offline replica."""
+
+
+@scenario("resize_offline.feature", "Expand an unpublished volume")
+def test_expand_an_unpublished_volume():
+    """Expand an unpublished volume."""
+
+
+@scenario(
+    "resize_offline.feature", "Expand an unpublished volume and then make it published"
+)
+def test_expand_an_unpublished_volume_and_then_make_it_published():
+    """Expand an unpublished volume and then make it published."""
+
+
+@scenario(
+    "resize_offline.feature", "Expand an unpublished volume with an offline replica"
+)
+def test_expand_an_unpublished_volume_with_an_offline_replica():
+    """Expand an unpublished volume with an offline replica."""
+
+
+@scenario("resize_offline.feature", "Shrink an unpublished volume")
+def test_shrink_an_unpublished_volume():
+    """Shrink an unpublished volume."""
+
+
+@given("a deployer cluster")
+def a_deployer_cluster():
+    """a deployer cluster."""
+
+
+@pytest.fixture(scope="module")
+def tmp_files():
+    files = []
+    for itr in range(DEFAULT_REPLICA_CNT + 1):
+        files.append(f"/tmp/node-{itr + 1}-disk")
+    yield files
+
+
+@pytest.fixture(scope="module")
+def disks(tmp_files):
+    for disk in tmp_files:
+        if os.path.exists(disk):
+            os.remove(disk)
+        with open(disk, "w") as file:
+            file.truncate(DEFAULT_POOL_SIZE)
+
+    # /tmp is mapped into /host/tmp within the io-engine containers
+    yield list(map(lambda file: f"/host{file}", tmp_files))
+
+    for disk in tmp_files:
+        if os.path.exists(disk):
+            os.remove(disk)
+
+
+# This fixture will be automatically used by all tests.
+# It starts the deployer which launches all the necessary containers.
+# Pools are created for convenience such that they are available for use by the tests.
+@pytest.fixture(autouse=True, scope="module")
+def init(disks):
+    Deployer.start(
+        4,
+        faulted_child_wait_period="5s",
+        reconcile_period="1s",
+        cache_period="1s",
```

Comment:
```suggestion
        faulted_child_wait_period="100ms",
        reconcile_period="100ms",
        cache_period="100ms",
```

Reply:
Kept faulted_child_wait_period at 2s to have `wait_faulted_child` check work reliably.